src/cgutils.cpp

```diff
@@ -1786,11 +1786,17 @@ static Value *emit_arrayflags(const jl_cgval_t &tinfo, jl_codectx_t *ctx)
 {
     assert(tinfo.isboxed);
     Value *t = tinfo.V;
+#ifdef STORE_ARRAY_LEN
+    int arrayflag_field = 2;
+#else
+    int arrayflag_field = 1;
+#endif
```

Comment:
using `offsetof(jl_array_t, flags) / sizeof(void*)` here instead can help avoid the magic constants